exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/IdTable.kt

```diff
@@ -37,19 +37,49 @@ abstract class IdTable<T : Comparable<T>>(name: String = "") : Table(name) {
 
     /** All base columns that make up this [IdTable]'s identifier column. */
     val idColumns = HashSet<Column<out Comparable<*>>>()
+        get() = field.validate()
+
+    private var idColumnsValidated: Boolean = false
+    private var inTableInitialization: Boolean = false
+
+    private fun <F : HashSet<Column<out Comparable<*>>>> F.validate(): F {
+        if (idColumnsValidated || inTableInitialization) return this
+
+        val primaryKeyColumns = primaryKey?.columns?.toSet().orEmpty()
+        return when {
+//            primaryKeyColumns.isEmpty() || minus(primaryKeyColumns).isNotEmpty() -> {
+//                val message = "Table definition must include a primary key with all id columns: ${joinToString { it.name }}"
+//                exposedLogger.error(message)
+//                error(message)
+//            }
+            this == primaryKeyColumns -> {
+                idColumnsValidated = true
+                this
+            }
+            else -> {
+                primaryKeyColumns.minus(this).forEach { missingColumn ->
```

Comment:
Above you told that:

> I thought the above logic made sense, that a table with an entityId() column must have it as a primary key, but there are a couple examples in our tests that do not follow this rule and do not have a primary key at all.

Which means that if the table has an entity id (or compsite id) it does not mean that table must have same primary key, sounds logical right.

But I want to make sure (I trust that you already checked it, but it does not looks obvious for me yet) that with the statement below you say, that if table has primary key, all columns of that primary key must be in the `idColumns`, is it so?

Reply:
Yes, I assumed that if an `IdTable` has a primary key defined, then all columns of that primary key must also be in `idColumns`.

Reply:
> But in the original PR, you made a comment that it should not be possible to create multiple entityId() on a non-CompositeIdTable, so I don't see how the first statement is logical.

Multiple `entityId` can be used only with `CompositeIdTable`, I didn't want to say that non-`CompositeIdTable` should be able to have more that one id column

The question was about validation. Two questions are:
- must all the `idColumns` columns be in `primaryKey`?
- must all the `primaryKey` columns be in `idColumns`?

Now from your comments I see that answer to the first one is No, and to the second one is Yes.